setup.py

```diff
@@ -12,18 +12,27 @@
 # language governing permissions and limitations under the License.
 from __future__ import absolute_import
 
-from glob import glob
 import os
+import re
+from glob import glob
 
 from setuptools import setup, find_packages
 
 
+def get_version():
```

Comment:
I believe we should follow pep-0396 guidance https://www.python.org/dev/peps/pep-0396/#classic-distutils :
In classic distutils, the simplest way to add the version string to the setup() function in setup.py is to do something like this:
```
from elle import __version__
setup(name='elle', version=__version__)
```
In the PEP author's experience however, this can fail in some cases, such as when the module uses automatic Python 3 conversion via the 2to3 program (because setup.py is executed by Python 3 before the elle module has been converted).

In that case, it's not much more difficult to write a little code to parse the __version__ from the file rather than importing it. Without providing too much detail, it's likely that modules such as distutils2 will provide a way to parse version strings from files. E.g.:
```
from distutils2 import get_version
setup(name='elle', version=get_version('elle.py'))
```

Reply:
Nevermind, from offline conversation that seems to be the new way (especially since Distutils2 development is stopped):
https://packaging.python.org/guides/single-sourcing-package-version/